salt/states/reg.py

```diff
@@ -617,3 +621,317 @@ def key_absent(name, use_32bit_registry=False):
+                return False
+    return True
+
+
+def _imported_do_import(reference_reg_file, use_32bit_registry):
+    r'''
+    This is a utility function used by imported. This function calls
+    the module function reg.import_file. If that function then throws
+    exceptions, this function will catch some of them.
+    The function returns an ordered pair (A,B).
+    The values are defined as follows
+       A: Whether or not the operation succeeded expressed as a boolean.
+       B: an error message if the operation failed. Otherwise None.
+    '''
+    try:
+        __salt__['reg.import_file'](reference_reg_file, use_32bit_registry)
+    except ValueError as err:
+        err_msg_fmt = "Call to module function 'reg.import_file' has failed. Error is '{0}'"
+        err_msg = comment_fmt.format(str(err))
+        return (False, err_msg)
+    except CommandExecutionError as err:
+        err_msg_fmt = "Call to module function 'reg.import_file' has failed. Error is '{0}'."
+        err_msg = comment_fmt.format(err.message)
+        return (False, err_msg)
+    return (True, None)
+    
+
+def _imported_get_change_requirements(reference_data):
+    r'''
+    This is a utility function used by imported. It's job is to
+    determine, using the the reference data, what
+    changes are required, and if changes are required an operation
+    which will effectuate the changes is also returned.
+    Specifically an ordered pair A, B is returned. A and B are defined
+    as follows.
+           A: If an error occurs the value of A will be False, otherwise True
+           B: If an error occurs the value will be an error message,
+              otherwise the value will be an ordered pair (c,d). The values
+              of c and d are defined as follows.
+                 c: This is a dictionary of required changes. It is not a dictionary
+                    of changes that have happened.
+                 d: If c is non-empty, d will be a function with no paramaters.
+                    This function is an operation which will effectuate the changes
+                    in the change dictionary.
+                    Id c is empty, d will just be None (no operation is needed when no
+                    changes are needed). Note that the change dictionary created here 
+                    is rather minimal and probably
+                    should be fleshed out to include specific changes.
+    '''
+    (reference_reg_file, reference_parse, use_32bit_registry) \
+        = reference_data
+    reg_location = (reference_parse.sections())[0]
+    try:
+        state_parse = _imported_state_data(reg_location, use_32bit_registry)
+    except CommandExecutionError as err:
+        comment = _imported_compose_cmd_execution_err_msg(err)
+        return (False, comment)
+    changes = {}
+    operation = None
+    if not _imported_test(state_parse, reference_data):
+        changes['old'] = "Registry unmodified."
```

Comment:
These should be empty.     The `comment `filed is the only field to have comments.  new and old should only contain what the old values & delete items and new contain changes to existing values & additional items.

Reply:
> These should be empty. The `comment `filed is the only field to have comments. new and old should only contain what the old values & delete items and new contain changes to existing values & additional items.

The changes dictionary i'm returning gets used when in test mode or later on if the operation is performed and succeeds (I hang on to to is so I can use it in the final output), so it needs to not be empty in the situations where i'm actually using it.

I realise that the contents of the dictionary isn't standards conformant. I can fix that but it'll be a few minutes of work since I really should be creating the changes dictionary as I'm testing. It's not a big deal.